files/edge-core/patches/0001-Patch-Broadcast-gateway-stats-to-LWM2M-resources.patch

```diff
@@ -0,0 +1,626 @@
++                                                    false);
++}
++
+ bool edgeclient_get_resource_value_and_attributes(const char *endpoint_name,
+                                                   const uint16_t object_id,
+                                                   const uint16_t object_instance_id,
+diff --git a/edge-client/gateway_services_resource.c b/edge-client/gateway_services_resource.c
+index 2bc3bbd..4257967 100644
+--- a/edge-client/gateway_services_resource.c
++++ b/edge-client/gateway_services_resource.c
+@@ -22,9 +22,24 @@
+ #include "edge-client/edge_client.h"
+ #include <stddef.h>
+ #include <string.h>
++#include <unistd.h>
+ #include "mbed-trace/mbed_trace.h"
+ #include "edge-client/edge_core_cb_result.h"
+ #include "edge-client/gateway_services_resource.h"
++#include <stdlib.h>
++
++#define GATEWAY_STATS_OBJ_ID 3
++
++#define GATEWAY_STATS_VERSION_RES_ID 4001
++#define GATEWAY_STATS_CPU_TEMP_RES_ID 3303
++#define GATEWAY_STATS_CPU_PCT_RES_ID 3320
++#define GATEWAY_STATS_RAM_FREE_RES_ID 3321
++#define GATEWAY_STATS_RAM_TOTAL_RES_ID 3322
++#define GATEWAY_STATS_DISK_FREE_RES_ID 3323
++#define GATEWAY_STATS_DISK_TOTAL_RES_ID 3324
++
++#define GATEWAY_STATS_BUFFER_SIZE 128
++#define GATEWAY_STATS_REFRESH_TIMER 10
+ 
+ static int gsr_add_service_instance(uint16_t obj_instance_id, uint8_t* id, uint32_t id_len, uint8_t* enabled, uint8_t* config, uint32_t config_len)
+ {
+@@ -137,4 +152,302 @@ void gsr_add_gateway_services_resource()
+     }
+ }
+ 
++//run shell cmd and return results to out_buffer
++int sys_exec(char* out_buffer, const char* cmd) {
++    FILE *fp;
++    char buffer[128];
++    fp = popen(cmd, "r");
++    if (fp == NULL)
++        return -1;
++    while (fgets(buffer, sizeof(buffer), fp) != NULL) {
++        strcat(out_buffer, buffer);
++    }
++    pclose(fp);
++    return 0;
++}
++
++//set a value in the cloud for obj_id/res_id to string result of the cmd shell command.
++void gsr_set_resource_helper(uint16_t obj_id, uint16_t res_id, const char* cmd)
++{
++    char result[GATEWAY_STATS_BUFFER_SIZE];
++    *result = 0;
++    int ret = sys_exec(result, cmd);
++    if (ret != 0)
++        strcpy(result, "-1");
++    pt_api_result_code_e eret = edgeclient_update_resource_value(NULL,
++                                                                 obj_id,
++                                                                 0,
++                                                                 res_id,
++                                                                 (uint8_t *) result,
++                                                                 strlen(result));
++    if (eret != PT_API_SUCCESS)
++        tr_debug("EdgeClient update resource /%d/0/%d failed with code: %d", obj_id, res_id, eret);
++}
++
++//set a value in the cloud for obj_id/res_id to nValue
++void gsr_set_resource_helper_int(uint32_t obj_id, uint16_t res_id, int value)
++{
++    static int32_t swap = 0;
```

Comment:
I think we still need a better understanding about what mbed-client does with this memory.  On cursory examination, it looks like it calls free() on the pointer, which would be bad for us since we're giving it a pointer to a static variable.